tests/testsuite/publish_lockfile.rs

```diff
@@ -492,3 +492,97 @@ fn ignore_lockfile_inner() {
         )
         .run();
 }
+
+#[cargo_test]
+fn respect_workspace_lockfile() {
+    // Issue #11148
+    Package::new("serde", "0.2.0").publish();
+    Package::new("serde", "0.2.1").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+                license = "MIT"
+                description = "foo"
+
+                [dependencies]
+                serde = "0.2"
+
+                [workspace]
+                members = ["bar"]
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                authors = []
+                license = "MIT"
+                description = "bar"
+                workspace = ".."
+
+                [dependencies]
+                serde = "0.2"
+            "#,
+        )
+        .file("bar/src/main.rs", "fn main() {}")
+        .build();
+    p.cargo("generate-lockfile").run();
+    p.cargo("update -p serde:0.2.1 --precise 0.2.0").run();
```

Comment:
Is it possible to move `Package::new("serde", "0.2.1").publish();` to this line (and to replace it)?